Sources/RSocketCore/Channel Handler/ConnectionStreamHandler.swift

```diff
@@ -14,10 +14,75 @@
  * limitations under the License.
  */
 
+import Foundation
 import NIO
 
-final class ConnectionStreamHandler: ChannelInboundHandler {
-    typealias InboundIn = FrameBody
-    typealias OutboundOut = FrameBody
-    // TODO
+final class ConnectionStreamHandler {
+    private var keepAliveHandle: RepeatedTask?
+    
+    /// receive time in **seconds** of the last keepalive frame
+    private var lastReceivedTime: TimeInterval
+    private var timeBetweenKeepaliveFrames: Int32
+    private var maxLifetime: Int32
+    private let connectionSide: ConnectionRole
+    /// returns the current time in **seconds**
+    private let now: () -> TimeInterval
+
+    init(timeBetweenKeepaliveFrames: Int32,
+         maxLifetime: Int32,
+         connectionSide: ConnectionRole,
+         now: @escaping () -> TimeInterval = { ProcessInfo.processInfo.systemUptime }
+    ) {
+        self.timeBetweenKeepaliveFrames = timeBetweenKeepaliveFrames
+        self.maxLifetime = maxLifetime
+        self.connectionSide = connectionSide
+        self.now = now
+        self.lastReceivedTime = now()
+    }
+}
+
+extension ConnectionStreamHandler: ChannelInboundHandler {
+    typealias InboundIn = Frame
+    typealias OutboundOut = Frame
+
+    func channelRead(context: ChannelHandlerContext, data: NIOAny) {
+        let frame = unwrapInboundIn(data)
+        switch frame.body {
+        case let .keepalive(body):
+            lastReceivedTime = now()
+            if body.respondWithKeepalive {
+                let keepAliveFrame = KeepAliveFrameBody(respondWithKeepalive: false, lastReceivedPosition: 0, data: Data()).asFrame()
```

Comment:
Let's document it with a comment that `lastReceivedPosition` is set to `0` statically because we don't support resumability yet:
```suggestion
                // we don't support resumability yet so we send always send lastReceivedPosition as 0
                let keepAliveFrame = KeepAliveFrameBody(respondWithKeepalive: false, lastReceivedPosition: 0, data: Data()).asFrame()
```